source/layers/validation/checkers/events_deadlock/zel_events_deadlock_checker.cpp

```diff
@@ -0,0 +1,256 @@
+    ze_kernel_handle_t hKernel,              ///< [in] handle of the kernel object
+    const ze_group_count_t *pLaunchFuncArgs, ///< [in] thread group launch arguments
+    ze_event_handle_t hSignalEvent,          ///< [in][optional] handle of the event to signal on completion
+    uint32_t numWaitEvents,                  ///< [in][optional] number of events to wait on before launching; must be 0
+                                             ///< if `nullptr == phWaitEvents`
+    ze_event_handle_t *phWaitEvents          ///< [in][optional][range(0, numWaitEvents)] handle of the events to wait
+                                             ///< on before launching
+) {
+    std::cout << "eventsDeadlockChecker: zeCommandListAppendLaunchKernelEpilogue" << std::endl;
+    printf("\tNumber of wait events: %d\n", numWaitEvents);
+    for (uint32_t i = 0; i < numWaitEvents; i++) {
+        printf("\tphWaitEvents[%d] pointer address: %p\n", i, phWaitEvents[i]);
+    }
+    // printf("\thEvent pointer address: %p\n", hEvent);
+
+    return ZE_RESULT_SUCCESS;
+}
+
+ze_result_t
+eventsDeadlockChecker::ZEeventsDeadlockChecker::zeCommandListAppendLaunchCooperativeKernelPrologue(
+    ze_command_list_handle_t hCommandList,   ///< [in] handle of the command list
+    ze_kernel_handle_t hKernel,              ///< [in] handle of the kernel object
+    const ze_group_count_t *pLaunchFuncArgs, ///< [in] thread group launch arguments
+    ze_event_handle_t hSignalEvent,          ///< [in][optional] handle of the event to signal on completion
+    uint32_t numWaitEvents,                  ///< [in][optional] number of events to wait on before launching; must be 0
+                                             ///< if `nullptr == phWaitEvents`
+    ze_event_handle_t *phWaitEvents          ///< [in][optional][range(0, numWaitEvents)] handle of the events to wait
+                                             ///< on before launching
+) {
+
+    return ZE_RESULT_SUCCESS;
+}
+
+ze_result_t
+eventsDeadlockChecker::ZEeventsDeadlockChecker::zeCommandListAppendLaunchCooperativeKernelEpilogue(
+    ze_command_list_handle_t hCommandList,   ///< [in] handle of the command list
+    ze_kernel_handle_t hKernel,              ///< [in] handle of the kernel object
+    const ze_group_count_t *pLaunchFuncArgs, ///< [in] thread group launch arguments
+    ze_event_handle_t hSignalEvent,          ///< [in][optional] handle of the event to signal on completion
+    uint32_t numWaitEvents,                  ///< [in][optional] number of events to wait on before launching; must be 0
+                                             ///< if `nullptr == phWaitEvents`
+    ze_event_handle_t *phWaitEvents          ///< [in][optional][range(0, numWaitEvents)] handle of the events to wait
+                                             ///< on before launching
+) {
+
+    return ZE_RESULT_SUCCESS;
+}
+
+ze_result_t
+eventsDeadlockChecker::ZEeventsDeadlockChecker::zeCommandListAppendMemoryCopyPrologue(
```

Comment:
I would be insterested in seeing a high level outline of how the algorithm works before getting to implementation of specific functions
1) Which APIs will add things to graph
2) Which APIs will remove things from graph
3) Which APIs will execute deadlock detection mechanism